plugin/system_copy.vim

```diff
@@ -33,23 +43,58 @@ function! s:resolve_mode(type, arg)
   endif
 endfunction
 
-function! s:CopyCommandForCurrentOS()
+function! s:currentOS()
+  if !empty(s:os)
+    return s:os
+  endif
   let os = substitute(system('uname'), '\n', '', '')
-  if has("gui_mac") || os == 'Darwin'
-    return 'pbcopy'
+  if has("gui_mac") || os ==? 'Darwin'
+    let s:os = s:mac
+    return s:mac
   elseif has("gui_win32")
+    let s:os = s:windows
+    return s:windows
+  elseif os ==? 'Linux'
+    let s:os = s:linux
+    return s:linux
+  endif
+  exe "normal \<Esc>"
+  throw "unknown OS: " . os
+endfunction
```

Comment:
I'm not sure I think it's worth the added complexity of caching the os name. Likewise, I try to keep to single return value from functions. Maybe use the following?

``` vim
function! s:currentOS()
  let os = substitute(system('uname'), '\n', '', '')
  if has("gui_mac") || os ==? 'Darwin'
    let known_os = s:mac
  elseif has("gui_win32")
    let known_os = s:windows
  elseif os ==? 'Linux'
    let known_os = s:linux
  else
    exe "normal \<Esc>"
    throw "unknown OS: " . os
  endif
  return known_os
endfunction
```